### PR TITLE
Xaiki/faster load

### PR DESCRIPTION
**Pull request by vankasteelj**

### CODE REVIEW

**Comment by xaiki** [comment]
@vankasteelj review plez ?

**Comment by vankasteelj** [comment]
When refreshing multiple times, I eventually get to this error:

```
Uncaught TypeError: this.ui.covers.append is not a function
    at n.<anonymous> (chrome-extension://afkpagbcdgaomobnegnmhijblpoeanna/src/app/lib/views/browser/item.js:132:32)
    at HTMLImageElement.posterCache.onload (chrome-extension://afkpagbcdgaomobnegnmhijblpoeanna/src/app/lib/views/browser/item.js:148:17)
```

**Comment by xaiki** [comment]
that is very weird... added a patch to check for this.

File: src/app/lib/models/content_item.js [hunk 1]
```diff
@@ -14,28 +14,13 @@
...
-                return;
-            }
-
-            if (info.overview) {
-                info.synopsis = info.overview;
-            }
-
-            return this.set(info);
-        },
-
```
Comment (xaiki) [inline src/app/lib/models/content_item.js]: whut ?

File: src/app/lib/models/content_item.js [hunk 2]
```diff
@@ -14,28 +14,13 @@
             this.set('providers', providers);
             this.updateHealth();
 
-            var id = attrs.imdb_id;
-            providers.metadata && providers.metadata.getMetadata(id)
-                .then(this.applyMetadata.bind(this))
-                .catch(e => console.error('error loading metadata', e));
+            !attrs.poster && providers.metadata && providers.metadata.getImages(attrs).then(this.set.bind(this)).catch(e => console.error('error loading metadata', e));
```
Comment (xaiki) [inline src/app/lib/models/content_item.js]: please keep at least the indent...

**Comment by xaiki** [comment]
so ? can this get in ?

File: src/app/lib/models/content_item.js [hunk 3]
```diff
@@ -14,7 +14,7 @@
             this.set('providers', providers);
             this.updateHealth();
 
-            !attrs.poster && providers.metadata &&
+            providers.metadata &&
```
Comment (vankasteelj) [inline src/app/lib/models/content_item.js]: this makes the fadein fail for me

Comment (xaiki) [inline src/app/lib/models/content_item.js]: yeah fadein is hacky, is there a better way ?

**Comment by vankasteelj** [comment]
the issue with calling 300 items instead of 50 isnt present in master. The issue is in here

**Comment by xaiki** [comment]
ok @vankasteelj a bunch of new fixes, good news: we have no more ghosts,
hasMore detection might be a bit broken, and we do call again into *LOADED* providers, but memoizee handles that and we shouldn't create huge amounts of elements anymore.

**Comment by vankasteelj** [comment]
There is an infinite loop somewhere. See the gif:
http://gph.is/2hGFUsv


(note: the issue with loading more than needed on scroll isnt resolved, but at least it's not actually making requests)

**Comment by xaiki** [comment]
@vankasteelj ??

**Comment by vankasteelj** [comment]
I still have an issue: 

```
app.js:702 TypeError: Cannot read property 'length' of undefined(…) "TypeError: Cannot read property 'length' of undefined
    at formatArchiveForButter (E:\GIT\butter-desktop\node_modules\butter-provider-archive\index.js:123:39)
    at Promise_then_fulfilled (E:\GIT\butter-desktop\node_modules\q\q.js:766:44)
    at Promise_done_fulfilled (E:\GIT\butter-desktop\node_modules\q\q.js:835:31)
    at Fulfilled_dispatch [as dispatch] (E:\GIT\butter-desktop\node_modules\q\q.js:1229:9)
    at Pending_become_eachMessage_task (E:\GIT\butter-desktop\node_modules\q\q.js:1369:30)
    at RawTask.call (E:\GIT\butter-desktop\node_modules\asap\asap.js:40:19)
    at flush (E:\GIT\butter-desktop\node_modules\asap\raw.js:50:29)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)"
```

other than that, awesome work

**Comment by vankasteelj** [comment]
well I shouldnt have merged. Details are broken.

**Pull request merged**